compiler-core/src/parse.rs

```diff
@@ -2795,37 +2803,53 @@ where
         }
     }
 
-    fn parse_attributes(&mut self, attributes: &mut Attributes) -> Result<(), ParseError> {
+    fn parse_attributes(
+        &mut self,
+        attributes: &mut Attributes,
+    ) -> Result<Option<SrcSpan>, ParseError> {
+        let mut start_of_attributes = None;
+        let mut end_of_attributes = None;
+
         while let Some((start, _)) = self.maybe_one(&Token::At) {
-            self.parse_attribute(start, attributes)?;
+            if start_of_attributes.is_none() {
+                start_of_attributes = Some(start);
+            }
+
+            end_of_attributes = Some(self.parse_attribute(start, attributes)?);
         }
-        Ok(())
+
+        Ok(start_of_attributes.map(|start| SrcSpan {
+            start,
+            end: end_of_attributes.expect(""),
```

Comment:
Do not use `.expect` if it can be avoided, and always give it a message.

The change above to use SrcSpan means this isn't needed 👍